Launching-new-features.md

```diff
@@ -38,43 +38,49 @@ Say you are working on a large scale user-facing feature that will take 1 or mor
+    - Once you have confirmation that the feature can be made permanent, merge one last PR to "un-gate" the feature and move the feature flag to the deprecated stage (one of the stages listed in `core/feature_flag_list.py`, meant for flags that are no longer in use). Additionally, in the same PR, please remove all remaining references to the feature flag from the codebase (for example, in all the `if` blocks you created to gate the feature).
 
 
 ## Follow the steps below to add a new feature flag
 
 ### Creating the flag
 
-1. Add a new unique feature flag name in the `PARAM_NAMES` enum class in `core/domain/platform_parameter_list.py`, similar to a key-value pair. Example:
+1. Add a new unique feature flag name in the `FeatureNames` enum class in `core/feature_flag_list.py`, similar to a key-value pair. Example:
 
 ```python
-class ParamNames(enum.Enum):
-    """Enum for parameter names."""
+class FeatureNames(enum.Enum):
+    """Enum for feature flag names."""
     // ...
     NEW_FEATURE = 'new_feature',
 
 ```
 
-2. Create and register a feature flag instance in the `core/domain/platform_parameter_registry.py` file with its name, description and stage (see the [Feature Stage](#feature-stage-explanation) section for more details). Example:
+2. Add the name of the new feature flag to one of the feature name lists (`DEV_FEATURES_LIST`, `TEST_FEATURES_LIST`, or `PROD_FEATURES_LIST`) in `core/feature_flag_list.py` according to its stage. Note: if the name is added to the wrong list, a backend test error will raise. Example:
 
 ```python
-Registry.create_feature_flag(
-    PARAM_NAMES.NEW_FEATURE,
-    'This is a newly created feature.',
-    platform_parameter_domain.FeatureStages.DEV,
-)
+DEV_FEATURES_LIST = [
+    # ...
+    FeatureNames.NEW_FEATURE,
+]
 ```
 
-3. Add the name of the newly created feature to one of the feature name lists (`DEV_FEATURES_LIST`, `TEST_FEATURES_LIST`, or `PROD_FEATURES_LIST`) in `core/platform_feature_list.py` according to its stage. Note: if the name is added to the wrong list, a backend test error will raise. Example:
+3. Add feature flag description and feature stage to the `FEATURE_FLAG_NAME_TO_DESCRIPTION_AND_FEATURE_STAGE` present in `core/feature_flag_list.py`.
+Example:
 
 ```python
-DEV_FEATURES_LIST = [
-    # ...
-    params.PARAM_NAMES.NEW_FEATURE,
-]
+FEATURE_FLAG_NAME_TO_DESCRIPTION_AND_FEATURE_STAGE = {
+  # ...
+  FeatureNames.NEW_FEATURE.value: (
+    (
+      'This is description for new feature flag.',
+      feature_flag_domain.ServerMode.DEV
+    )
+  ),
+  # ...
+}
 ```
 
-4. To make the feature flag available in the front-end, you need to add it into the name enum in `core/templates/domain/platform_feature/feature-status-summary.model.ts` as well:
+4. To make the feature flag available in the front-end, you need to add it into the name enum in `core/templates/domain/feature-flag/feature-status-summary.model.ts` as well:
```

Comment:
Should we say the FeatureNames enum rather than the name enum?

Reply:
Done